.ci/updateStackReleaseVersion.groovy

```diff
@@ -0,0 +1,146 @@
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import groovy.transform.Field
+
+@Library('apm@master') _
+
+// To store the next and current release versions
+@Field def releaseVersions
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-pipeline-library'
+    ORG_NAME = 'elastic'
+    HOME = "${env.WORKSPACE}"
+    NOTIFY_TO = credentials('notify-to')
+    PIPELINE_LOG_LEVEL = 'INFO'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        git(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken', url: "https://github.com/${ORG_NAME}/${REPO}.git")
+      }
+    }
+    stage('Fetch latest versions') {
+      steps {
+        echo ' TODO: calculate the versions'
+      }
+    }
+    stage('Send Pull Request'){
+      steps {
+        createPullRequest(repo: env.REPO,
+                          branchName: 'master',
+                          labels: 'automation',
+                          reviewer: 'elastic/observablt-robots-on-call',
+                          title: '[automation] Update Elastic stack release version source of truth')
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def createPullRequest(Map args = [:]) {
+  def message = createPRDescription(latestVersions)
+  prepareContext(repo: args.repo, branchName: args.branchName)
+  if (args?.stackVersions?.size() == 0) {
+    error('createPullRequest: stackVersions is empty. Review the artifacts-api for the branch ' + args.branchName)
+  }
+  writeFile file: 'resources/versions/releases.properties', text: """current_6=${releaseVersions.get('current.6')}
+current_7=${releaseVersions.get('current.7')}
+next_minor_7=${releaseVersions.get('next.minor.7')}
+next_patch_7=${releaseVersions.get('next.patch.7')}"""
```

Comment:
to avoid having to change the configuration across reops every major, we can use something more neutral like:
* maintenance -> 6.x.x
* current -> 7.x.x
* current_next_minor -> 7.x.0
* current_next_patch -> 7.Y.x

Reply:
I initially though to use something like that, but as long as the consumers use the same branch strategy (`6.<minor>`, `7.<minor>` or `7.x`) then the `current.7`/`current.6` could be easy to manage since the `<minor>` is explicitly in the branch name.

For instance:

- Given beats@7.x then I'd expect to use `next_minor_7` = 7.14.0

Therefore `next_minor_${branch.split('\.')[0]}` should be the way to know the value in the consumers

- Given beats@7.13 then I'd expect to use `next_patch_7` = 7.13.3

- Given beats@6.9 then I'd expect to use `current_6` = 6.8.15

I think the match will be easier when using the `<minor>` in the property name since there is no need to match other naming with the branch strategy.

What do you think?